app/components/new-course.js

```diff
@@ -1,80 +1,76 @@
-import Component from '@ember/component';
-import { reads } from '@ember/object/computed';
+import Component from '@glimmer/component';
+import { tracked } from "@glimmer/tracking";
+import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import moment from 'moment';
-import { validator, buildValidations } from 'ember-cp-validations';
-import ValidationErrorDisplay from 'ilios-common/mixins/validation-error-display';
-import { task } from 'ember-concurrency';
-
-const Validations = buildValidations({
-  currentSchool: validator('presence', true),
-  selectedYear: validator('presence', true),
-  title: [
-    validator('presence', true),
-    validator('length', { min: 3, max: 200 })
-  ]
-});
+import { validatable, Length, NotBlank } from 'ilios-common/decorators/validation';
+import { dropTask, restartableTask } from 'ember-concurrency-decorators';
 
 const THIS_YEAR = parseInt(moment().format('YYYY'), 10);
 const YEARS = [
-  THIS_YEAR-2,
-  THIS_YEAR-1,
+  THIS_YEAR - 2,
+  THIS_YEAR - 1,
   THIS_YEAR,
-  THIS_YEAR+1,
-  THIS_YEAR+2
+  THIS_YEAR + 1,
+  THIS_YEAR + 2
 ];
 
-export default Component.extend(Validations, ValidationErrorDisplay, {
-  intl: service(),
-  store: service(),
-
-  classNames: ['new-course'],
+@validatable
+export default class NewCourseComponent extends Component {
+  @service intl;
+  @service store;
+  @service iliosConfig;
 
-  currentSchool: null,
-  currentYear: null,
-  selectedYear: null,
-  title: null,
-  cancel() {},
-  save() {},
-  years: Object.freeze(YEARS),
+  @tracked @NotBlank() selectedYear;
+  @tracked @NotBlank() @Length(3, 200) title;
+  @tracked years = Object.freeze(YEARS);
```

Comment:
This `Object.freeze` shouldn't be necessary anymore since this component is now a JS class and they don't share state the way ember object based components did.

Reply:
:heavy_check_mark: